src/main/java/graphql/validation/rules/UniqueDirectiveNamesPerLocation.java

```diff
@@ -57,17 +58,23 @@ public void checkOperationDefinition(OperationDefinition operationDefinition) {
     }
 
     private void checkDirectivesUniqueness(Node<?> directivesContainer, List<Directive> directives) {
-        Set<String> names = new LinkedHashSet<>();
-        directives.forEach(directive -> {
-            String name = directive.getName();
-            if (names.contains(name)) {
+        Set<String> directiveNames = new LinkedHashSet<>();
+
+        for (Directive directive : directives) {
+            String directiveName = directive.getName();
+            GraphQLDirective graphQLDirective = getValidationContext().getSchema().getDirective(directiveName);
+
+            if (graphQLDirective == null) continue;
+            if (graphQLDirective.getDefinition() != null && graphQLDirective.getDefinition().isRepeatable()) continue;
```

Comment:
I think we will want to transfer the AST isRepeatable into the runtime `GraphQLDirective` class.

That is inside SchemaGenerator related code we transfer these booleans over

The AST elements are for reading SDL - the GraphqlXXX class are aimed as the runtime equivalents with more logic in them and more state rules.

Reply:
thanks for your guidance